internal/routing/ip.go

```diff
@@ -15,7 +17,15 @@ var (
 	errInterfaceIPNotFound = errors.New("IP address not found for interface")
 )
 
-func (r *Routing) assignedIP(interfaceName string) (ip net.IP, err error) {
+func ipMatchesFamily(ip net.IP, family int) bool {
+	if (family == netlink.FAMILY_V6 && ip.To4() == nil) ||
+		(family == netlink.FAMILY_V4 && ip.To4() != nil) {
+		return true
+	}
+	return false
```

Comment:
nit can be simplified as:

```suggestion
	return (family == netlink.FAMILY_V6 && ip.To4() == nil) ||
		(family == netlink.FAMILY_V4 && ip.To4() != nil)
```